src/contracts/core/Slasher.sol

```diff
@@ -5,233 +5,475 @@ import "./SlasherStorage.sol";
 
 import "../libraries/EpochUtils.sol";
 import "../libraries/SlashingAccountingUtils.sol";
+import "../libraries/EIP1271SignatureUtils.sol";
 
 /**
  * @notice Slasher
  */
 contract Slasher is SlasherStorage {
+
+    uint256 constant MAGNITUDE_ALLOCATION_DELAY = 21 days;
+    uint256 constant MAGNITUDE_DEALLOCATION_DELAY = 21 days;
+    uint256 constant MAX_PENDING_MAGNITUDE_UPDATES = 3;
+
+    /**
+     * @notice this struct is used in MagnitudeAdjustmentParam in order to specify
+     * an operator's slashability for a certain operator set
+     * @param operatorSet the operator set to change slashing parameters for
+     * @param magnitudeDiff magnitude difference; the difference in proportional parts of the operator's slashable stake
+     * that the operator set is getting. This struct is used either in allocating or deallocating
+     * slashable stake to an operator set. Slashable stake for an operator set is
+     * (slashableMagnitude / sum of all slashableMagnitudes for the strategy/operator + nonSlashableMagnitude) of
+     * an operator's delegated stake.
+     */
+    struct OperatorSetMagnitudeParam {
```

Comment:
can u use the interface pls

Reply:
https://github.com/Layr-Labs/protocol-designs/pull/12/files#diff-ba96d21a88b00975a5f3e56038db353ef3764544cb431fcbae024e9726ba74f3R17-R20
Its up-to-date according to latest gitbook status